### PR TITLE
feat: add lse return on trtllm-gen attention

### PR DESCRIPTION
**Pull request by yyihuang**
## 🔍 Related Issues

https://github.com/sgl-project/sglang/issues/9471

## 🚀 Pull Request Checklist

Thank you for contributing to FlashInfer! Before we review your pull request, please make sure the following items are complete.

### ✅ Pre-commit Checks

- [ ] I have installed `pre-commit` by running `pip install pre-commit` (or used your preferred method).
- [ ] I have installed the hooks with `pre-commit install`.
- [ ] I have run the hooks manually with `pre-commit run --all-files` and fixed any reported issues.

> If you are unsure about how to set up `pre-commit`, see [the pre-commit documentation](https://pre-commit.com/).

## 🧪 Tests

- [ ] Tests have been added or updated as needed.
- [ ] All tests are passing (`unittest`, etc.).

### CODE REVIEW

**Comment by zhyncs** [comment]
@yzh119 @yyihuang what's the progress about this pr

**Comment by zhyncs** [comment]
Can we merge this before v0.3.0 @yzh119

**Comment by yzh119** [comment]
@yyihuang fyi, https://github.com/flashinfer-ai/flashinfer/pull/1566/commits/c5822e78d4237879fbda66f745056e9469a6f1c6 should make bf16 mla tests passed (but unfortunately, fp8 UT will fail).

**Comment by yzh119** [comment]
https://github.com/flashinfer-ai/flashinfer/pull/1566/commits/91e3b832eaa53ad48e22ce3db67e988af1c91aa1 this commit should fix fp8.

The fundamental reason is because trtllm kernels use internal scale for rowsum/rowmax which might not align with provided bmm scale. For fp8, the reason we need to `- log2(448)` is that the logits is multipled with 448 (maximum value of fp8 e4m3) to improve numerical stability.

**Comment by pavanimajety** [comment]
@yyihuang Is there a plan to revive this PR?? @yzh119

**Comment by yyihuang** [comment]
> @yyihuang Is there a plan to revive this PR?? @yzh119

We can move it to https://github.com/flashinfer-ai/flashinfer/pull/2332.
